aws/ec2.go

```diff
@@ -82,60 +82,75 @@ func getNamedSG(name, environment, profile, sshUser string, ports ...int64) (sg
-					for _, iprange := range ipperm.IpRanges {
-						if iprange.CidrIp == nil ||
-							iprange.Description == nil ||
-							*iprange.Description == "" ||
-							*iprange.Description != sshUser {
-							continue
-						}
-
-						// add this CIDR to this SG for this `port`
-						sg.portToMyIPs[port] = append(
-							sg.portToMyIPs[port],
-							*iprange.CidrIp,
-						)
+			// ensure `iprange` is for `sshUser`
+			for _, iprange := range ipperm.IpRanges {
+				// skip unless Description is sshUser
+				if iprange.CidrIp == nil ||
+					iprange.Description == nil ||
+					*iprange.Description == "" ||
+					*iprange.Description != sshUser {
+					continue
+				}
+
+				// see if `ipperm` is for current allowed `ports` for this SG
+				isUnusualPort := true
+				for _, port := range ports {
+					if *ipperm.ToPort == port {
+						isUnusualPort = false
+						break
 					}
-					break
 				}
+				if isUnusualPort {
+					out.Highlight(out.WARN, "%s has unexpected port %s for %s", name, *ipperm.ToPort, sshUser)
+				}
+				// add CIDR to list that is keyed on ToPort
+				sg.portToMyIPs[*ipperm.ToPort] = append(
+					sg.portToMyIPs[*ipperm.ToPort],
+					*iprange.CidrIp,
+				)
 			}
 		}
 	}
 
 	return
 }
 
-func getBastionSGForEnvironment(environment, profile, sshUser string) (secGroup, error) {
-	return getNamedSG(environment+" - bastion", environment, profile, sshUser, 22, 443)
+func getBastionSGForEnvironment(environment, profile, sshUser string, morePorts []int64) (secGroup, error) {
+	return getNamedSG(
+		environment+" - bastion", environment, profile, sshUser,
+		append(morePorts, 22, 443),
+	)
 }
 
-func getELBPublishingSGForEnvironment(environment, profile, sshUser string) (secGroup, error) {
-	return getNamedSG(environment+" - publishing elb", environment, profile, sshUser, 80, 443)
+func getELBPublishingSGForEnvironment(environment, profile, sshUser string, morePorts []int64) (secGroup, error) {
+	return getNamedSG(
+		environment+" - publishing elb", environment, profile, sshUser,
+		append(morePorts, 443),
+	)
 }
 
-func getELBWebSGForEnvironment(environment, profile, sshUser string) (secGroup, error) {
-	return getNamedSG(environment+" - web elb", environment, profile, sshUser, 80, 443)
+func getELBWebSGForEnvironment(environment, profile, sshUser string, morePorts []int64) (secGroup, error) {
+	return getNamedSG(
+		environment+" - web elb", environment, profile, sshUser,
+		append(morePorts, 80, 443),
+	)
 }
 
 func getConcourseWebSG(sshUser string) (secGroup, error) {
-	return getNamedSG("concourse-ci-web", "", "", sshUser, 443)
+	return getNamedSG("concourse-ci-web", "", "", sshUser, []int64{443})
 }
 
 // AllowIPForEnvironment adds your IP to this environment
-func AllowIPForEnvironment(sshUser, environment, profile string) error {
-	return changeIPsForEnvironment(true, sshUser, environment, profile)
+func AllowIPForEnvironment(sshUser, environment, profile string, morePorts config.AddPorts) error {
```

Comment:
Just wondering if it's worth making the names consistent. 
- Variable uses 'morePorts'
- Type uses `AddPorts`
- Config field uses `ExtraPorts`

Reply:
ugh, yes, now you've pointed it out